### DISCUSSION TITLE
0.6.x Failed to create internal tables on Cassandra

### DISCUSSION THREAD

**Discussion by StardustXiaoT** [JanusGraph/janusgraph#3038] [Q&A] [answered]
- Version: 0.6.1
- Storage Backend: Cassandra cluster
- Mixed Index Backend: none
- Link to discussed bug: similar issue https://lists.lfaidata.foundation/g/janusgraph-users/topic/unconfigured_table_exceptions/85741983?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,0,85741983,previd=1634849714017760765,nextid=1631631322121685927&previd=1634849714017760765&nextid=1631631322121685927
- Expected Behavior: janusgraph able to create tables on cassnadra automatically 
- Current Behavior: sometime it does not auto create internal tables and sometime only create few. need manual create on table.
- Steps to Reproduce: 

Create and run Janusgraph 0.6.1 binary on docker. 
Bitnami cassandra cluster as back-end storage.
cassandra:
  enabled: true
  image:
    registry: docker.io
    repository: bitnami/cassandra
    tag: 4.0.3-debian-10-r8
  replicaCount: 3

Two method to create graph: 
 1. default cql properties with fixed keyspace param. 
 2. ConfiguredGraphFactory.create("graph1");

Both methods may trouble creating tables on initializing keyspace on Cassandra.

**Comment by li-boxuan** [comment]
Did you try adding the config `storage.cql.request-timeout = 10000` (10 seconds)? By default, it's 2 seconds, and some users have reported problems with 2 seconds being too low.

**Comment by StardustXiaoT** [comment]
> 
I found this on my configure. I believe I tried this before.
#storage.cql.request-timeout = 24000

**Comment by li-boxuan** [comment]
Have you also tried the `advanced.metadata.schema.debouncer.window` config as mentioned in https://lists.lfaidata.foundation/g/janusgraph-users/topic/unconfigured_table_exceptions/85741983 ?

**Comment by StardustXiaoT** [comment]
> 

    #storage.cql.internal.string-configuration = datastax-java-driver { advanced.metadata.schema.debouncer.max-events = 20 }
    #storage.cql.internal.string-configuration = datastax-java-driver { advanced.metadata.schema.debouncer.window = 1 seconds }
    #storage.cql.request-timeout = 24000

yep I tried

**Comment by StardustXiaoT** [comment, marked as answer]
I did some test on versions and clustering on cassandra, here is some result. 
ConfiguredGraphFactory

janusgraph 0.6.1 cassandra 3.11.12 single node(docker) immediate creation on internal tables
janusgraph 0.6.1 cassandra 4.0.3 single node(docker) immediate creation on internal tables

janusgraph 0.6.1 cassandra 3.11.12 cluster(k8s) exceed evaluationTimeout 30000, changes to 60000, then it works
janusgraph 0.5.3 cassandra 3.11.12 cluster(k8s) exceed evaluationTimeout 30000, changes to 60000, then it works

**Reply by li-boxuan** [reply]
Cool, I presume you have figured out the appropriate configurations by yourself. If you don't mind I am marking your own reply as the answer.
